R/pd_deflation.R

```diff
@@ -0,0 +1,574 @@
+
+      if (length(dt_c[, unique(reporting_level)]) > 1)  { # Needed??
+
+        dt_c <- adjust_population(dt_c, pop)
+
+      }
+
+      ### Format vars  ---------
+
+      char_to_fct(dt_c)
+    },
+
+    error = function(cnd){
+
+      survey_id <- c(.logenv$survey_id)
+
+      cli::cli_alert("The survey {survey_id} was skipped")
+
+      log_failure(cnd)
+
+      NA
+
+    }
+  )
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  # Return   ---------
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  return(dt_f)
+
+}
+
+#' Deflation of welfare for group data
+#'
+#' @inheritParams deflation
+#' @return data.table
+#' @export
+deflation.pipgd <- function(dt,  cpi, ppp, pop,...) {
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  # computations   ---------
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  # Defenses (Make copy -> need to fix)
+  fnms <- names(formals())
+  fnms <- fnms[!fnms %in% "..."]
+
+  for (i in seq_along(fnms)) {
+    rr <- get(fnms[[i]])
+    if (inherits(rr, "data.table")) {
+      assign(fnms[i], copy(rr))
+
+    } else {
+      assign(fnms[i], qDT(rr))
+    }
+
+  }
+
+  ### Small fix ---------
+  dt_c <- copy(dt)
+
+  assign("survey_id",
+         attributes(dt_c)$survey_id$values,
+         envir = .logenv)
+
+  # on.exit ------------
+  on.exit({
+    rm(survey_id,
+       envir = .logenv)
+  })
+
+  dt_f <- tryCatch(
+    expr = {
+      #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      ## Add reporting level --------
+
+      dt_c <- add_rep_lvl(dt_c) # Maybe not needed with Tefera new aux data.
+
+      #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      ## Deflation --------
+
+      ### Merge survey with ppp and cpi ---------
+
+      dt_c <- add_aux(dt_c, ppp, cpi)
+
+      ### Welfare LCU ---------
+
+      dt_c <- welfare_lcu(dt_c)
+
+      ### Delfate welfare vector ------
+
+      dt_c <- deflate_wlf(dt_c)
+
+      ### Format vars  ---------
+
+      char_to_fct(dt_c)
+    },
+
+    error = function(cnd){
+
+      survey_id <- c(.logenv$survey_id)
+
+      cli::cli_alert("The survey {survey_id} was skipped")
+
+      log_failure(cnd)
+
+      NA
+
+    }
+  )
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  # Return   ---------
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  return(dt_f)
+
+}
+
+
+#' Convert PPP data from `pipload` to wide format
+#'
+#' @param ppp data frame with ppp data from `pipload::pip_load_aux("ppp")`
+#'
+#' @return data.table with PPP values to wide format based on versioning
+#' @export
+#'
+#' @examples
+#' ppp <-  pipload::pip_load_aux("ppp")
+#' x   <-  ppp_to_wide(ppp)
+#' names(x)
+ppp_to_wide <- function(ppp) {
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ## Identify ppp versions --------
+
+  ppp[,
+      ppp_version := {
+        x <- paste0("ppp_", ppp_year, "_", release_version, "_", adaptation_version)
+        x <- gsub("_v", "_0", x )
+      }
+  ]
+
+  ppp_v <- ppp[, unique(ppp_version)]
```

Comment:
We already have some code that make this kind of transformation. This should no be necessary. Rossana's file should be easy to manipulate. This format is not sustainable... Let's talk about it.